src/main/java/com/epam/preprod/sirenko/filesearcher/ChainBuilder.java

```diff
@@ -0,0 +1,181 @@
+package com.epam.preprod.sirenko.filesearcher;
+
+import com.epam.preprod.sirenko.filesearcher.filters.*;
+import com.epam.preprod.sirenko.util.ConsoleReader;
+import com.epam.preprod.sirenko.util.PrintToConsole;
+
+import java.io.IOException;
+import java.sql.Timestamp;
+
+public class ChainBuilder {
+	private static final String INCORRECT_INPUT_NO_FILTERS_CHOSEN = "You haven't choose any filter. Try again.";
+	private static final String INCORRECT_INPUT_ONE_OR_ZERO = "Please print only 0 or 1";
+	private static final String INCORRECT_INPUT_SIZE_RANGE = "You've entered wrong format of size range";
+	private static final String INCORRECT_INPUT_SIZE_RANGE_FIRST_BIGGER = "You've entered wrong size range - first is bigger then second";
+	private static final String INCORRECT_INPUT_EDIT_DATE_RANGE = "You've entered wrong format of dates range";
+	private static final String INCORRECT_INPUT_EDIT_DATE_RANGE_FIRST_LATER = "You've entered wrong dates - first is not earlier then second";
+	private static final String FIRST_QUESTION = "Search by file name? Print 0 for no, 1 for yes";
+	private static final String FIRST_QUESTION_PRINT_NAME = "Print file name";
+	private static final String SECOND_QUESTION = "Search by file extension? Print 0 for no, 1 for yes";
+	private static final String SECOND_QUESTION_PRINT_EXTENSION = "Print file extension";
+	private static final String THIRD_QUESTION = "Search by file size range? Print 0 for no, 1 for yes";
+	private static final String THIRD_QUESTION_PRINT_SIZE = "Print size range in kilobytes in format 85 230";
+	private static final String FOURTH_QUESTION = "Search by file edit date range? Print 0 for no, 1 for yes";
+	private static final String FOURTH_QUESTION_PRINT_DATE = "Print file edit dates in format 2021-09-10 13:45:00 2021-09-23 13:45:00";
+	private Filter filter;
+	
+	public Filter builder() throws IOException {
+		return firstInputRequest();
+	}
+	
+	private Filter firstInputRequest() throws IOException {
+		PrintToConsole.printString(FIRST_QUESTION);
+		String inputString = ConsoleReader.readFromConsole();
```

Comment:
If the user accidentally enters a string from the console, what happens?

Reply:
user will see message "Please print only 0 or 1" (line 35), check of input is in line 34 - if (!checkInputStringIsNumber(inputString) {